packages/liveblocks-react-lexical/src/comments/floating-threads.tsx

```diff
@@ -0,0 +1,321 @@
+  }, [editor, handleUpdateRange]);
+
+  const handleEscapeKeydown = useCallback((): boolean => {
+    if (range === null) return false;
+    setRange(null);
+    return true;
+  }, [range]);
+
+  useEffect(() => {
+    return editor.registerCommand(
+      KEY_ESCAPE_COMMAND,
+      handleEscapeKeydown,
+      COMMAND_PRIORITY_HIGH
+    );
+  }, [editor, handleEscapeKeydown]);
+
+  const isCollapsed = useIsSelectionCollapsed();
+
+  if (range === null || isCollapsed === null || !isCollapsed) return null;
+
+  return (
+    <FloatingThreadPortal
+      range={range.range}
+      container={document.body}
+      {...divProps}
+    >
+      {range.threads.map((thread) => (
+        <ThreadWrapper
+          key={thread.id}
+          thread={thread}
+          Thread={Thread}
+          onEscapeKeydown={handleEscapeKeydown}
+          className="lb-lexical-floating-threads-thread"
+        />
+      ))}
+    </FloatingThreadPortal>
+  );
+}
+
+interface FloatingThreadPortalProps
+  extends Omit<HTMLAttributes<HTMLDivElement>, "children"> {
+  range: Range;
+  container: HTMLElement;
+  children: ReactNode;
+}
+
+export const FLOATING_THREAD_COLLISION_PADDING = 10;
+
+function FloatingThreadPortal(props: FloatingThreadPortalProps) {
+  const { container, range, children, className, style, ...divProps } = props;
+
+  const {
+    refs: { setReference, setFloating },
+    strategy,
+    x,
+    y,
+  } = useFloating({
+    strategy: "absolute",
+    placement: "bottom",
+    middleware: [
+      flip({ padding: FLOATING_THREAD_COLLISION_PADDING, crossAxis: false }),
+      offset(10),
+      hide({ padding: FLOATING_THREAD_COLLISION_PADDING }),
+      shift({
+        padding: FLOATING_THREAD_COLLISION_PADDING,
+        limiter: limitShift(),
+      }),
+      size({
+        padding: FLOATING_THREAD_COLLISION_PADDING,
+        apply({ availableWidth, availableHeight, elements }) {
+          elements.floating.style.setProperty(
+            "--lb-lexical-floating-threads-available-width",
+            `${availableWidth}px`
+          );
+          elements.floating.style.setProperty(
+            "--lb-lexical-floating-threads-available-height",
+            `${availableHeight}px`
+          );
+        },
+      }),
+    ],
+    whileElementsMounted: (...args) => {
+      return autoUpdate(...args, {
+        animationFrame: true,
+      });
+    },
+  });
+
+  useLayoutEffect(() => {
+    setReference({
+      getBoundingClientRect: () => range.getBoundingClientRect(),
+    });
+  }, [setReference, range]);
+
+  return createPortal(
+    <div
+      ref={setFloating}
+      {...divProps}
+      style={{
+        ...style,
+        position: strategy,
+        top: 0,
+        left: 0,
+        transform: `translate3d(${Math.round(x)}px, ${Math.round(y)}px, 0)`,
+        minWidth: "max-content",
+      }}
+      className={classNames(
+        "lb-root lb-portal lb-elevation lb-lexical-floating-threads",
+        className
+      )}
```

Comment:
The container that holds floating threads is labelled with classname `lb-lexical-floating-threads`.